library/alloc/src/boxed.rs

```diff
@@ -239,7 +239,9 @@ pub struct Box<
 /// This is the surface syntax for `box <expr>` expressions.
 #[rustc_intrinsic]
 #[unstable(feature = "liballoc_internals", issue = "none")]
-pub fn box_new<T>(x: T) -> Box<T>;
+pub fn box_new<T>(x: T) -> Box<T> {
+    Box::write(Box::new_uninit(), x)
+}
```

Comment:
This shouldn't have any effect. `box_new` is already getting replaced when building MIR, but fallback bodies are only used during codegen/const-eval which operate after building MIR.

https://github.com/rust-lang/rust/blob/0c4fa2690de945f062668acfc36b3f8cfbd013e2/compiler/rustc_mir_build/src/thir/cx/expr.rs#L387-L404